src/formatDate.js

```diff
@@ -8,7 +8,55 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[fromFormat.length - 1];
+  const joinElement = toFormat[toFormat.length - 1];
+
+  const arrOfDate = date.split(splitElement);
+  const newDateArr = [];
+
+  const varDay = 'DD';
+  const varMonth = 'MM';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === varDay) {
+      const index = toFormat.indexOf(varDay);
+      const day = arrOfDate[i];
+
+      newDateArr[index] = day;
+    } else if (fromFormat[i] === varMonth) {
+      const index = toFormat.indexOf(varMonth);
+      const month = arrOfDate[i];
+
+      newDateArr[index] = month;
+    } else {
+      const fromFormatYear = fromFormat[i];
+      let year = arrOfDate[i];
+
+      for (let j = 0; j < toFormat.length - 1; j++) {
+        if ((toFormat[j] !== varDay) & (toFormat[j] !== varMonth)) {
+          const toFormatYear = toFormat[j];
+
+          const index = j;
+
+          if (fromFormatYear.length > toFormatYear.length) {
+            year = year.slice(-2);
```

Comment:
When slicing the year to convert from 'YYYY' to 'YY', there is no check to ensure that 'year' has at least 4 characters. While the input is assumed to be correct, adding a length check would make the code more robust.